src/impls/$select-by/$select-by.js

```diff
@@ -0,0 +1,30 @@
+import { $async, $await } from '../../../generate/async.macro.cjs';
+
+import { $iterableCurry } from '../../internal/$iterable.js';
+import { $__peekerate } from '../$peekerate/$peekerate.js';
+
+$async;
+export function $__selectBy(iterable, selector, mapper) {
+  const peekr = $await($__peekerate(iterable));
+
+  if (!peekr.done) {
+    let bestItem = peekr.value;
```

Comment:
I've explicitly avoided using the term "item" in my code and documentation. I believe `bestItem` should be called `value` and `bestValue` should be `mappedValue`.